### PR TITLE
`ultralytics 8.0.228` add training `time` argument

### LINKED ISSUE #6919: Include training time arg
## Summary
Two possible routes to use. 
1. **Primary:** Estimate the average duration required per training epoch to update LR-scheduler to finalize training on epoch number that is closest to the `time` provided.

    - EX:
        `yolo train time=3600 # arg example` 
        - if first 5 epochs average ~5 minutes/epoch, update LR-scheduler to finish at epoch `12` instead of default `100`

1. **Secondary:** Use `time` argument to 'interrupt' training (same as `CTRL+C`) after `t` training duration. **The model at this stage will be incomplete and NOT ready for use.** The goal with this method is to allow for a type of orchestration, where the model can be trained for an extended amount of time in smaller chunks if compute resources or availability are constrained.

    - see #6750 for additional details and conversation

## Requirements
- minimal modifications to training code
- if possible, only add single argument (maybe only possible per method)
- where feasible, reuse `EarlyStopping` class

### PR DESCRIPTION
**Pull request by glenn-jocher**
@kalenmike PR for timed training, work in progress.

copilot:all

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
Enhanced Ultralytics documentation and added a time-based training termination.

### 📊 Key Changes
- Improved clarity in instance segmentation and object tracking documentation.
- Introduced a new `time` parameter for training configuration to specify the maximum training duration.
- Modified worker allocation logic to prevent potential worker over-allocation.
- Applied code structure improvements for better maintainability.
- Bumped Ultralytics version to 8.0.228.

### 🎯 Purpose & Impact
- 📝 Documentation improvements make it easier for users to understand instance segmentation features.
- 🕔 The `time` parameter offers users the ability to set a fixed training duration, leading to predictable training times and better resource management (helpful when using cloud-based services with time constraints).
- 💻 Code enhancements contribute to a more robust and efficient codebase, potentially reducing bugs and improving user experience.
- 🆙 Version increment signifies ongoing development and feature additions, signaling to users that the project is actively maintained.

### CODE REVIEW

**Comment by glenn-jocher** [comment]
@kalenmike @Burhan-Q @Laughing-q ok I think this is ready! I tested with this command. LR successfully updated to target minimum final LR on final time, and `epochs` update over training to estimated number of epochs that can be run in given `time`. In this example `epochs=1000` is ignored since `time=0.05` (hours) is set.

I think I might redefine time to seconds though to be consistent with the python time package. This might result in some really big numbers though, i.e. to train for 24 hours `time=86400`.

```bash
yolo train device=4 time=0.05 data=coco8.yaml batch=1 project=test1 epochs=1000 patience=100
```


Screenshot 2023-12-19 at 22 27 59

**Comment by kalenmike** [comment]
@glenn-jocher Awesome!
